src/SleepRepo.js

```diff
@@ -9,21 +9,43 @@ class SleepRepo {
       return totalQuality;
     }, 0) / this.sleepData.length).toFixed(1));
   }
-
-  returnAboveAverageSleepers(week) {
-    let dataByUser = this.sleepData.reduce((arr, user) => {
+  returnDataByUser() {
+    return this.sleepData.reduce((arr, user) => {
       if (!arr[user.userID - 1]) {
         arr[user.userID - 1] = [user];
       } else {
         arr[user.userID - 1].push(user);
       }
       return arr;
     }, []);
+  }
 
-    let avgSleepQualityPerUser = dataByUser.map(user => [...user].splice(-7 * week, 7)).map(user => user.reduce((totalQuality, day) => {
-      totalQuality += day.sleepQuality;
+  returnAvgSleepDataPerUser(week, relevantProperty) {
+    let dataByUser = this.returnDataByUser()
+
+    return dataByUser.map(user => [...user].splice(-7 * week, 7)).map(user => user.reduce((totalQuality, day) => {
+      totalQuality += day[relevantProperty];
       return totalQuality;
-    }, 0)).map(user => Number((user / 7).toFixed(2)));
+    }, 0))
+  }
+
+  returnAboveAverageSleepers(week) {
+    let dataByUser = this.returnDataByUser();
+    // let dataByUser = this.sleepData.reduce((arr, user) => {
+    //   if (!arr[user.userID - 1]) {
+    //     arr[user.userID - 1] = [user];
+    //   } else {
+    //     arr[user.userID - 1].push(user);
+    //   }
+    //   return arr;
+    // }, []);
+
+    // let avgSleepQualityPerUser = dataByUser.map(user => [...user].splice(-7 * week, 7)).map(user => user.reduce((totalQuality, day) => {
+    //   totalQuality += day.sleepQuality;
+    //   return totalQuality;
+    // }, 0)).map(user => Number((user / 7).toFixed(2)));
+
+    let avgSleepQualityPerUser = this.returnAvgSleepDataPerUser(dataByUser, 'sleepQuality').map(user => Number((user / 7).toFixed(2)));
```

Comment:
The first argument of returnAvgSleepDataPerUser dataByUser is supposed to be a number.  I think you use week in other methods.